src/Parser.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+namespace CSVParser;
+
+class Parser {
+
+    var $path;
+    var $header_array = array();
+    var $delimiter;
+
+    function __construct ($path, $header_array = array(), $delimiter = ",") {

```

Comment:
Since you are specifying default values here you don't need to specify initial values above. It is also recommended you use single quotes for strings in PHP unless string interpolation is needed. I.e. this would change the method signature to 
```php
public function __construct($path, $headers = []), $delimiter = ',')
```

We can talk about string interpolation later but a simple example is
```php
$greeting = 'Hello';
echo "{$greeting} World!";

// Output
//    Hello World!
```

Reply:
Also noticed I changed `$headers` to use the simpler [] declaration for an empty array. This is just stylistic and not a required change.

Reply:
Another thing I did not realize was an option but I'll use it for simplicity.